std/math.c3

```diff
@@ -0,0 +1,9 @@
+@require_once "std/type_traits.c3"
+
+function [T] abs : (value : T) -> TypeIf<T, IsIntegral<T>> = {
+    if value < 0 {
+        return -value;
+    } else {
+        return value;
+    }
```

Comment:
That was a good read, I've had fun playing around with it :smiley:. I do think is a premature optimization tho.

1) LLVM at -O1 is correctly able to deduce both algorithms are an `abs` (compiling them to the same IR call). The only difference is that there are MORE debug symbols in the branchless version since more variables are collapsed (it will be slower if anything).
![image](https://github.com/DaveDuck321/GrapheneLang/assets/30243291/13a34fb1-1cc0-404e-b33f-760cbd2cd41e)


2) The AMD64 output actually more closely resembles the version WITH branches: LLVM is able to outsmart us, see what we're trying to do with the bit-twiddling, and remove our optimization.
![image](https://github.com/DaveDuck321/GrapheneLang/assets/30243291/a892c255-96ee-4894-aaf9-fac7fb61d749)

Let's just keep the branches for readability, LLVM can see it's an `abs` so it'll codegen the optimal assembly for each target regardless.

Reply:
Hah, fair enough. Kinda amazed that it can see straight through the bit-hacks!

Interesting to note that for riscv, the assembly output is essentially the "patented version" from that website